runner/compute-driver/local/compute-execute

```diff
@@ -0,0 +1,136 @@
+
+    # show configuration
+    echo " output_relation_tmp=$output_relation_tmp"
+    echo
+
+    # use an empty temporary table as a sink instead of TRUNCATE'ing the output_relation
+    deepdive-initdb "$output_relation"
+    deepdive-sql "
+        DROP TABLE IF EXISTS $output_relation_tmp;
+        CREATE TABLE $output_relation_tmp (LIKE $output_relation);
+    "
+fi
+
+# set up named pipes for parallel processes and make sure they are cleaned up upon exit
+[[ -z $input_sql       ]] || for i in $(seq $num_processes); do rm -f process-$i.input ; mkfifo process-$i.input ; done
+[[ -z $output_relation ]] || for i in $(seq $num_processes); do rm -f process-$i.output; mkfifo process-$i.output; done
+trap 'rm -f process-*.{input,output}' EXIT
+# now spawn processes attached to the named pipes in reverse order (from sink to source)
+
+if [[ -n $output_relation ]]; then
+    # use mkmimo again to merge outputs of multiple processes into a single stream
+    mkmimo process-*.output \> /dev/stdout |
+    # load the output data to the temporary table in the database
+    # XXX hiding default progress bar from deepdive-load
+    # TODO abbreviate this env into a show_progress option, e.g., recursive=false
+    show_progress input_to "$DEEPDIVE_CURRENT_PROCESS_NAME output" -- \
+    env DEEPDIVE_PROGRESS_FD=2 \
+    deepdive-load "$output_relation_tmp" /dev/stdin &
+fi
+
+# spawn multiple processes attached to the pipes
+if [[ -n $output_relation && -n $input_sql ]]; then # process with input from/output to database
+    for i in $(seq $num_processes); do
+        DEEPDIVE_CURRENT_PROCESS_INDEX=$i \
+        "$SHELL" -c "$command" <process-$i.input >process-$i.output &
+    done
+elif [[ -n $input_sql ]]; then # input-only process
+    for i in $(seq $num_processes); do
+        DEEPDIVE_CURRENT_PROCESS_INDEX=$i \
+        "$SHELL" -c "$command" <process-$i.input &
+    done
+elif [[ -n $output_relation ]]; then # output-only process
+    for i in $(seq $num_processes); do
+        DEEPDIVE_CURRENT_PROCESS_INDEX=$i \
+        "$SHELL" -c "$command" >process-$i.output &
+    done
+else # neither output_relation nor input_sql specified
+    for i in $(seq $num_processes); do
+        DEEPDIVE_CURRENT_PROCESS_INDEX=$i \
+        "$SHELL" -c "$command" &
+    done
+fi
+
+if [[ -n $input_sql ]]; then
+    # unload data from the database and pour into the pipes
+    show_progress output_from "$DEEPDIVE_CURRENT_PROCESS_NAME input" -- \
+    deepdive-sql eval "$input_sql" format="$DEEPDIVE_LOAD_FORMAT" |
+    # use mkmimo to distribute input data to multiple processes
+    mkmimo /dev/stdin \> process-*.input &
+fi
+
+wait  # until everything is done ##############################################
+
+if [[ -n $output_relation ]]; then
+    # rename the new temporary table
+    # TODO maybe use PostgreSQL's schema support here?
+    echo "Replacing $output_relation with $output_relation_tmp"
```

Comment:
Makes sense. Though the `+=` syntax in ddlog may give people the false impression that the output would be appended...